test/interchain/packet_forward_test.go

```diff
@@ -0,0 +1,192 @@
+package interchain
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"cosmossdk.io/math"
+	"github.com/celestiaorg/celestia-app/test/interchain/chainspec"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	"github.com/strangelove-ventures/interchaintest/v6"
+	"github.com/strangelove-ventures/interchaintest/v6/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v6/ibc"
+	"github.com/strangelove-ventures/interchaintest/v6/testreporter"
+	"github.com/strangelove-ventures/interchaintest/v6/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+type PacketMetadata struct {
+	Forward *ForwardMetadata `json:"forward"`
+}
+
+type ForwardMetadata struct {
+	Receiver       string        `json:"receiver"`
+	Port           string        `json:"port"`
+	Channel        string        `json:"channel"`
+	Timeout        time.Duration `json:"timeout"`
+	Retries        *uint8        `json:"retries,omitempty"`
+	Next           *string       `json:"next,omitempty"`
+	RefundSequence *uint64       `json:"refund_sequence,omitempty"`
+}
+
+// TestPacketForwardMiddleware verifies that Packet Forward Middleware (PFM) works as expected by sending a transfer from Celestia to Cosmos Hub, back to Celestia which then forwards it to Cosmos Hub 2. Celestia -> CosmosHub -> Celestia -> CosmosHub2 .
```

Comment:
[question] can this be simplified to CosmosHub -> Celestia -> CosmosHub2?

Basically it reduces one extra hop. And AFAICT we only care about verifying that Celestia can forward the metadata along

Reply:
I was doing that in order to get some tia on cosmos and also test forwarding all in one tx